test/cxx/Core/ControllerTest.cpp

```diff
@@ -53,21 +54,43 @@ namespace tut {
 				//   what's going on.
 				assert(mockSession != nullptr || mockException != nullptr);
 				callback(mockSession, mockException);
-				mockSession.reset();
-				mockException.reset();
+				if (--mockedSessionCount == 0) {
+					mockSession.reset();
+					mockException.reset();
+				}
+			}
+
+			virtual int ioConditions() override {
+				if (forceAsyncToWait) {
+					return EV_READ;
+				} else {
+					return Controller::ioConditions();
+				}
+			}
+
+			virtual double sessionSocketConnectTimeout(Request *req) override {
+				if (forceAsyncToWait) {
+					return req->connectTimeout / 1000.0;
+				} else {
+					return Controller::sessionSocketConnectTimeout(req);
+				}
 			}
 
 		public:
 			ApplicationPool2::AbstractSessionPtr mockSession;
 			ApplicationPool2::ExceptionPtr mockException;
+			unsigned int mockedSessionCount;
+			bool forceAsyncToWait;
```

Comment:
I think this name can be better. If we consider the goal of this flag, how about "forceSessionSocketConnectTimeout".

Reply:
I suppose, originally I had hoped to be able to allow the connect later (for additional test scenarios that would enable) which wouldn't fit with `forceSessionSocketConnectTimeout` but since that's not possible with this impl, I guess it fits.